fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/BindExpression.java

```diff
@@ -633,7 +639,31 @@ private Plan bindProject(MatchingContext<LogicalProject<Plan>> ctx) {
                 if (!excepts.isEmpty()) {
                     slots = Utils.filterImmutableList(slots, slot -> !boundExcepts.get().contains(slot));
                 }
-                boundProjections.addAll(slots);
+                if (!replaces.isEmpty()) {
+                    final Map<String, NamedExpression> replaceMap = new HashMap<>();
+                    final Set<String> replaced = new HashSet<>();
+                    for (NamedExpression replace : boundReplaces.get()) {
+                        Preconditions.checkArgument(replace instanceof Alias);
+                        Alias alias = (Alias) replace;
+                        replaceMap.put(alias.getName(), alias);
+                    }
+
+                    for (Slot slot : slots) {
+                        if (replaceMap.containsKey(slot.getName())) {
```

Comment:
what will happen if query like `select * replace ('123' as c1) from t, t t2` when t has columns `c1`?
i think we should thrown exception same as `select c1 from t, t t2`;

so the better way to do it may be we should use alias name as UnboundSlot, and use them to bind with project's child's output. and then we use `org.apache.doris.nereids.util.ExpressionUtils#replace` to replace project list.